### ISSUE TITLE
Installer from git is failing

### ISSUE THREAD

**Issue by stbenjam** [openshift-metal3/dev-scripts#839]
**Describe the bug**
When building the installer from source, install fails:

```
level=info msg="Creating infrastructure resources..."
level=debug msg="Symlinking plugin terraform-provider-local src: \"/home/notstack/go/src/github.com/openshift/installer/bin/openshift-install\" dst: \"/tmp/openshift-install-552856644/plugins
/terraform-provider-local\""
level=debug msg="Symlinking plugin terraform-provider-openstack src: \"/home/notstack/go/src/github.com/openshift/installer/bin/openshift-install\" dst: \"/tmp/openshift-install-552856644/plu
gins/terraform-provider-openstack\""
level=debug msg="Symlinking plugin terraform-provider-azureprivatedns src: \"/home/notstack/go/src/github.com/openshift/installer/bin/openshift-install\" dst: \"/tmp/openshift-install-5528566
44/plugins/terraform-provider-azureprivatedns\""
level=debug msg="Symlinking plugin terraform-provider-google src: \"/home/notstack/go/src/github.com/openshift/installer/bin/openshift-install\" dst: \"/tmp/openshift-install-552856644/plugin
s/terraform-provider-google\""
level=debug msg="Symlinking plugin terraform-provider-ignition src: \"/home/notstack/go/src/github.com/openshift/installer/bin/openshift-install\" dst: \"/tmp/openshift-install-552856644/plug
ins/terraform-provider-ignition\""
level=debug msg="Symlinking plugin terraform-provider-ironic src: \"/home/notstack/go/src/github.com/openshift/installer/bin/openshift-install\" dst: \"/tmp/openshift-install-552856644/plugin
s/terraform-provider-ironic\""
level=debug msg="Symlinking plugin terraform-provider-libvirt src: \"/home/notstack/go/src/github.com/openshift/installer/bin/openshift-install\" dst: \"/tmp/openshift-install-552856644/plugi
ns/terraform-provider-libvirt\""
level=debug msg="Symlinking plugin terraform-provider-random src: \"/home/notstack/go/src/github.com/openshift/installer/bin/openshift-install\" dst: \"/tmp/openshift-install-552856644/plugin
s/terraform-provider-random\""
level=debug msg="Symlinking plugin terraform-provider-aws src: \"/home/notstack/go/src/github.com/openshift/installer/bin/openshift-install\" dst: \"/tmp/openshift-install-552856644/plugins/t
erraform-provider-aws\""
level=debug msg="Symlinking plugin terraform-provider-azurerm src: \"/home/notstack/go/src/github.com/openshift/installer/bin/openshift-install\" dst: \"/tmp/openshift-install-552856644/plugi
ns/terraform-provider-azurerm\""
level=debug msg="Initializing modules..."
level=debug msg="2019/10/30 16:30:05 [DEBUG] Module installer: begin bootstrap"
level=debug msg="2019/10/30 16:30:05 [DEBUG] Module installer: bootstrap installed at "
level=debug msg="- bootstrap in "
level=debug msg="2019/10/30 16:30:05 [DEBUG] Module installer: begin masters"
level=debug msg="2019/10/30 16:30:05 [DEBUG] Module installer: masters installed at "
level=debug msg="- masters in "
level=error
level=error msg="Error: Unreadable module directory"
level=error
level=error msg="Unable to evaluate directory symlink: lstat ../tmp: no such file or directory"
level=error
level=error
level=error msg="Error: Failed to read module directory"
level=error
level=error msg="Module directory  does not exist or cannot be read."
level=error
```

**To Reproduce**

```
export KNI_INSTALL_FROM_GIT=true
```

**Expected/observed behavior**

Installer runs and completes when built from source

**Additional context**

**Comment by stbenjam** [comment]
This happened after the move to go 1.12.

The installer sets up terraform in a directory under /tmp, and puts modules in `/tmp/openshift-install-XXXXXXX/.tf/modules`.

I did notice the relative path in modules.json looks like it's off by one level:

```
[notstack@master-2 modules]$ pwd
/tmp/openshift-install-986401260/.tf/modules
[notstack@master-2 modules]$ cat modules.json 
{"Modules":[{"Key":"","Source":"","Dir":"../../../tmp/openshift-install-986401260"},{"Key":"bootstrap","Source":"./bootstrap","Dir":""},{"Key":"masters","Source":"./masters","Dir":""}]}[notstack@master-2 modules]$ ls ../../../tmp/openshift-install-986401260
ls: cannot access '../../../tmp/openshift-install-986401260': No such file or directory
[notstack@master-2 modules]$ ls ../../../../tmp/openshift-install-986401260
bootstrap  config.tf  main.tf  masters  OWNERS  plugins  terraform.baremetal.auto.tfvars.json  terraform.tfvars.json  variables-baremetal.tf  versions.tf
[notstack@master-2 modules]$ 
```

**Comment by hardys** [comment]
Hmm I'm not entirely sure where that broken relative path is coming from?

The plugins are an absolute symlink (so the openshift-install binary?!):

```
[shardy@master-1 plugins]$ ls -ltr
total 0
lrwxrwxrwx. 1 shardy shardy 72 Nov  1 13:13 terraform-provider-random -> /home/shardy/go/src/github.com/openshift/installer/bin/openshift-install
lrwxrwxrwx. 1 shardy shardy 72 Nov  1 13:13 terraform-provider-openstack -> /home/shardy/go/src/github.com/openshift/installer/bin/openshift-install
lrwxrwxrwx. 1 shardy shardy 72 Nov  1 13:13 terraform-provider-local -> /home/shardy/go/src/github.com/openshift/installer/bin/openshift-install
lrwxrwxrwx. 1 shardy shardy 72 Nov  1 13:13 terraform-provider-libvirt -> /home/shardy/go/src/github.com/openshift/installer/bin/openshift-install
lrwxrwxrwx. 1 shardy shardy 72 Nov  1 13:13 terraform-provider-ironic -> /home/shardy/go/src/github.com/openshift/installer/bin/openshift-install
lrwxrwxrwx. 1 shardy shardy 72 Nov  1 13:13 terraform-provider-ignition -> /home/shardy/go/src/github.com/openshift/installer/bin/openshift-install
lrwxrwxrwx. 1 shardy shardy 72 Nov  1 13:13 terraform-provider-google -> /home/shardy/go/src/github.com/openshift/installer/bin/openshift-install
lrwxrwxrwx. 1 shardy shardy 72 Nov  1 13:13 terraform-provider-azurerm -> /home/shardy/go/src/github.com/openshift/installer/bin/openshift-install
lrwxrwxrwx. 1 shardy shardy 72 Nov  1 13:13 terraform-provider-azureprivatedns -> /home/shardy/go/src/github.com/openshift/installer/bin/openshift-install
lrwxrwxrwx. 1 shardy shardy 72 Nov  1 13:13 terraform-provider-aws -> /home/shardy/go/src/github.com/openshift/installer/bin/openshift-install
[shardy@master-1 plugins]$ pwd
/tmp/openshift-install-082687517/plugins
```

But then the error comes from inside the code for `terraform init` where evidently something is creating a relative path, not spotted exactly where yet

**Comment by stbenjam** [comment]
Seems like the underlying cause is the golang role in metal3-dev-env installs go 1.12.0, it doesn't grab the latest. https://github.com/metal3-io/metal3-dev-env/pull/122 sets it to be 1.12.12, so it should fix CI at least.

It won't update /usr/local/go if it's older, you'll have to delete the directory before running dev-scripts again.  I'll have a look at a better fix

**Comment by hardys** [comment]
Closed via  metal3-io/metal3-dev-env#122 and  metal3-io/metal3-dev-env#123
